### PR TITLE
Add glacier service version and account id customizations

### PR DESCRIPTION
**Pull request by JordonPhillips**
This adds customizations for setting the default account id and setting the glacier service version

### CODE REVIEW

File: service/glacier/internal/customizations/apiversion.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+const glacierAPIVersionHeaderKey = "X-Amz-Glacier-Version"
+
+// AddGlacierAPIVersionMiddleware explicitly add handling for the Glacier api version
+// middleware to the operation stack.
+func AddGlacierAPIVersionMiddleware(stack *middleware.Stack, apiVersion string) {
+	stack.Serialize.Add(&GlacierAPIVersionMiddleware{apiVersion: apiVersion}, middleware.Before)
+}
+
+// GlacierAPIVersionMiddleware handles automatically setting Glacier's API version header.
+type GlacierAPIVersionMiddleware struct{
```
Comment (skotambkar) [inline service/glacier/internal/customizations/apiversion.go:20]: `GlacierAPIVersionMiddleware` doesn't need to be exported

File: service/glacier/internal/customizations/accountid.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+// AddDefaultAccountIDMiddleware adds the DefaultAccountIDMiddleware to the stack using
+// the given options.
+func AddDefaultAccountIDMiddleware(stack *middleware.Stack, setDefaultAccountID setDefaultAccountID) {
+	stack.Initialize.Add(&DefaultAccountIDMiddleware{
+		setDefaultAccountID: setDefaultAccountID,
+	}, middleware.Before)
+}
+
+// DefaultAccountIDMiddleware sets the account ID to "-" if it isn't already set
+type DefaultAccountIDMiddleware struct {
```
Comment (skotambkar) [inline service/glacier/internal/customizations/accountid.go:19]: `DefaultAccountIDMiddleware` should be unexported

File: service/glacier/api_client.go [hunk 3]
```diff
@@ -14,6 +14,7 @@ import (
 )
 
 const ServiceID = "Glacier"
+const ServiceAPIVersion = "2012-06-01"
```
Comment (skotambkar) [inline service/glacier/api_client.go:17]: We probably won't need `ServiceAPIVersion` exported

**Review by skotambkar — CHANGES_REQUESTED** [review]
Minor comments on not exporting middlewares var/constant

File: service/glacier/api_client.go (hunk 3 above)
Comment (JordonPhillips) [reply service/glacier/api_client.go:17]: This is generated by smithy-go

Comment (jasdel) [reply service/glacier/api_client.go:17]: It was added in another PR to smithy-go. Not sure benefit to user so much but useful for debugging potentially.  Is this an attribute that should be included in the user agent for the client? or does the request already capture that value?

**Review by jasdel — APPROVED** [review]

**Review by skotambkar — APPROVED** [review]

**Pull request merged**